app/jobs/remove_old_throttles_job.rb

```diff
@@ -41,4 +43,17 @@ def perform(now, limit: 1000, total_limit: 100_000)
       break if total_removed >= total_limit
     end
   end
+
+  def write_transaction_with_timeout(rails_env = Rails.env)
+    # rspec-rails's use_transactional_tests does not seem to act as expected when switching
+    # connections mid-test, so we just skip for now :[
+    return yield if rails_env.test?
+
+    ActiveRecord::Base.transaction do
+      # 30 seconds
+      quoted_timeout = ActiveRecord::Base.connection.quote(30_000)
+      ActiveRecord::Base.connection.execute("SET LOCAL statement_timeout = #{quoted_timeout}")
+      yield
+    end
```

Comment:
do we need to `ensure` and set the timeout back to the default value? I forget the scope of these